nix/machines/default.nix

```diff
@@ -0,0 +1,68 @@
+{
+  self,
+  lib,
+  ...
+}:
+let
+  allEnvironmentPaths = lib.filter (path: lib.hasPrefix "example-setup-" (builtins.baseNameOf path)) (
+    lib.attrValues (builtins.readDir ../test-environments)
+  );
```

Comment:
That's a mouthful, might benefit from a `lib.pipe` or a `|>`-chain

Reply:
we're not leaving in the nix future yet